### DISCUSSION TITLE
[V5] Auth - Frontend app on a different domain

### DISCUSSION THREAD

**Discussion by travis-r6s** [adonisjs/core#1911] [Help] [answered]
I have been reading through the docs, and came across this section: https://preview.adonisjs.com/guides/auth/api-guard#frontend-app-on-a-different-domain

Am I correct in thinking that that means I can use sessions + cookies, and send the auth cookie in an `Authorization` header if necessary?
If so, how should this be configured? I have tried various ways, like `Authorization: Bearer s%3AeyJtZXNzYW...`, `Authorization: s%3AeyJtZXNzYW...` etc, but those didn't seem to work.

Be glad of any pointers...

**Comment by Banashek** [comment]
Check the gif in [the auth section](https://preview.adonisjs.com/guides/auth/api-guard#authentication)

**Reply by travis-r6s** [reply]
Thanks - that looks like it is using an opaque token from the `api` method though, whereas the docs say the below about cookies:

> If your frontend application is running on a different domain, then you can still leverage the use of cookies by creating a proxy server. This server will proxy the request of your frontend application to your API transforming the cookie into the Authorization header.

**Reply by travis-r6s** [reply]
Or does it mean that the opaque token could be stored inside a cookie?

**Comment by RomainLanz** [comment, marked as answer]
Hey @thetre97! 👋 

You can (and should) use a Cookie if your application (frontend) and your server is a the same (sub)domain.

i.e Something like `example.com` and `api.example.com`

If you cannot have this architecture, I'd recommend creating a proxy. The proxy will be responsible to retrieve the API Token from the API and inject it inside a Cookie for your frontend.

**Reply by travis-r6s** [reply]
Thanks @RomainLanz.

I was using sessions with Nuxt + @nuxt/http, but found that it didn't always seem to pass the cookies when making a request server side, so requests returned as unauthorised. Not sure whether it was a mistake in my setup, or whether that package just doesn't support that - don't know whether you have any ideas there?

Either way, I found that I could store the API token as a cookie, then retrieve it server side, and set it as an `authorization` header, which works perfectly.
